src/main/java/vg/civcraft/mc/civchat2/CivChat2Manager.java

```diff
@@ -346,6 +368,22 @@ public void sendGroupMsg(Player sender, Group group, String message) {
 				sender.sendMessage(String.format(ChatStrings.globalMuted, TextUtil.formatDuration(mutedUntil - System.currentTimeMillis())));
 				return;
 			}
+			if(slowmodeTime != 0 && !sender.hasPermission("civchat2.globalslowmode")) {
+				boolean slowmode = instance.getCivChat2SettingsManager().getChatSlowmodeTimer().getValue(sender);
+				long ticks = slowmodeTime * 20;
+				if(!slowmode) {
```

Comment:
That setting should just be a long storing when the player last sent a message, that way all of this becomes easier and faster.

Reply:
Also generally try to "fail fast", meaning returning as early as possible and stacking logic to as little depth as possible. Instead of 
```
				if(!slowmode) {
					instance.getCivChat2SettingsManager().getChatSlowmodeTimer().setValue(sender, true);
					Bukkit.getScheduler().scheduleSyncDelayedTask(CivChat2.getInstance(), new Runnable() {
						@Override
						public void run() {
							instance.getCivChat2SettingsManager().getChatSlowmodeTimer().setValue(sender, false);
						}
					}, ticks);
				} else {
					sender.sendMessage(ChatStrings.globalSlowmode);
					return;
				}
```

```
				if(slowmode) {
					sender.sendMessage(ChatStrings.globalSlowmode);
					return;
				}
        instance.getCivChat2SettingsManager().getChatSlowmodeTimer().setValue(sender, true);
			  Bukkit.getScheduler().scheduleSyncDelayedTask(CivChat2.getInstance(), new Runnable() {
						@Override
						public void run() {
							instance.getCivChat2SettingsManager().getChatSlowmodeTimer().setValue(sender, false);
						}
				}, ticks);
```

is easier to read and expand upon